modules/workload-identity/main.tf

```diff
@@ -19,9 +19,23 @@ locals {
   gcp_sa_email            = google_service_account.cluster_service_account.email
 
   # This will cause terraform to block returning outputs until the service account is created
-  k8s_given_name       = var.k8s_sa_name != null ? var.k8s_sa_name : var.name
-  output_k8s_name      = var.use_existing_k8s_sa ? local.k8s_given_name : kubernetes_service_account.main[0].metadata[0].name
-  output_k8s_namespace = var.use_existing_k8s_sa ? var.namespace : kubernetes_service_account.main[0].metadata[0].namespace
+  k8s_given_name         = var.k8s_sa_name != null ? var.k8s_sa_name : var.name
+  output_k8s_name        = var.use_existing_k8s_sa ? local.k8s_given_name : kubernetes_service_account.main[0].metadata[0].name
+  output_k8s_namespace   = var.use_existing_k8s_sa ? var.namespace : kubernetes_service_account.main[0].metadata[0].namespace
+  token                  = var.use_existing_k8s_sa ? data.google_client_config.default.0.access_token : ""
+  cluster_ca_certificate = var.use_existing_k8s_sa ? data.google_container_cluster.primary.0.master_auth.0.cluster_ca_certificate : ""
+  cluster_endpoint       = var.use_existing_k8s_sa ? "https://${data.google_container_cluster.primary.0.endpoint}" : ""
+}
+
+data "google_container_cluster" "primary" {
```

Comment:
Should we use the auth submodule?

Reply:
I decided against the auth submodule because we instantiate this optionally if `use_existing_k8s_sa` is true.